src/main/java/io/debezium/connector/vitess/connection/VitessReplicationConnection.java

```diff
@@ -168,6 +174,15 @@ private int getNumOfRowEvents(Vtgate.VStreamResponse response) {
                 }
                 return num;
             }
+
+            // If we write to an internal database in Vitess for operation purpose
+            // Vitess will stream out an empty event with [BEGIN, VGTID, COMMIT].
+            private boolean isEmptyTransaction(Vtgate.VStreamResponse response) {
```

Comment:
As I mentioned in the PR comment, consider making it configurable and log the skipped events.